vocode/streaming/synthesizer/polly_synthesizer.py

```diff
@@ -0,0 +1,106 @@
+            raise Exception(
+                "Sampling rate not supported by AWS Polly",
+                synthesizer_config.sampling_rate,
+            )
+
+        self.sampling_rate = synthesizer_config.sampling_rate
+        self.client = client
+        self.language_code = synthesizer_config.language_code
+        self.voice_id = synthesizer_config.voice_id
+        self.thread_pool_executor = ThreadPoolExecutor(max_workers=1)
+
+    def synthesize(self, message: str) -> Any:
+        # Perform the text-to-speech request on the text input with the selected
+        # voice parameters and audio file type
+        return self.client.synthesize_speech(
+            Text=message,
+            LanguageCode=self.language_code,
+            TextType="text",
+            OutputFormat="pcm",
+            VoiceId=self.voice_id,
+            SampleRate=str(self.sampling_rate),
+        )
+
+    async def create_speech(
+        self,
+        message: BaseMessage,
+        chunk_size: int,
+        bot_sentiment: Optional[BotSentiment] = None,
+    ) -> SynthesisResult:
+        create_speech_span = tracer.start_span(
+            f"synthesizer.{SynthesizerType.POLLY.value.split('_', 1)[-1]}.create_total",
+        )
+        response = await asyncio.get_event_loop().run_in_executor(
+            self.thread_pool_executor, self.synthesize, message.text
+        )
+        audio_stream = response.get("AudioStream")
+        create_speech_span.end()
+
+        async def chunk_generator(audio_data_stream, chunk_transform=lambda x: x):
+            audio_buffer = await asyncio.get_event_loop().run_in_executor(
+                self.thread_pool_executor,
+                lambda: audio_stream.read(chunk_size),
+            )
+            if len(audio_buffer) != chunk_size:
+                yield SynthesisResult.ChunkResult(chunk_transform(audio_buffer), True)
+                return
+            else:
+                yield SynthesisResult.ChunkResult(chunk_transform(audio_buffer), False)
+            while True:
+                audio_buffer = audio_stream.read(chunk_size)
+                if len(audio_buffer) != chunk_size:
+                    yield SynthesisResult.ChunkResult(
+                        chunk_transform(audio_buffer[: len(audio_buffer)]), True
+                    )
+                    break
+                yield SynthesisResult.ChunkResult(chunk_transform(audio_buffer), False)
+
+        if self.synthesizer_config.should_encode_as_wav:
+            output_generator = chunk_generator(
+                audio_stream,
+                lambda chunk: encode_as_wav(chunk, self.synthesizer_config),
+            )
+        else:
+            output_generator = chunk_generator(audio_stream)
+
+        return SynthesisResult(
+            output_generator,
+            lambda seconds: self.get_message_cutoff_from_voice_speed(
+                message,
+                seconds,
+                60,  # value chosen completely arbitrarily
```

Comment:
This is the hard part about implementing polly synthesizer, as the API provides no good indicator of word boundaries and there is no guarantee on voice speed (see https://docs.aws.amazon.com/polly/latest/dg/voice-speed-vip.html)

Reply:
nevermind, [this ](https://docs.aws.amazon.com/polly/latest/dg/using-speechmarks.html)seems promising

Reply:
done